### ISSUE TITLE
Define violation messages for all violations.

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#15456]
We Should actually demand message to be always specified. (even Check has only one message)

I more and more inclined to think that it will be better to have violation message at all cases to let reader not rely on memory what it should be. In this case I completely forgot what this Check do, it required me jump  out of this page to get to know what this violation mean.

Code:
`if (o instanceof ColoredPoint(@SuppressWarnings("")int x, int y, int z)) { } // violation`

Preferable:
```
 // violation below 'The warning '''' cannot be suppressed at this location.'
if (o instanceof ColoredPoint(@SuppressWarnings("")int x, int y, int z)) { }
```

please share your thoughts, and if you both feel the same we can start following this rule.

I see this as closed https://github.com/checkstyle/checkstyle/issues/11214 ,  we can start update of all violation comments to have message.

We need to remove suppression list that we introduced at https://github.com/checkstyle/checkstyle/pull/15452/files

**Comment by mahfouz72** [comment]
@romani please add a good first issue if you agree

**Comment by romani** [comment]
yes, please contribute

**Comment by chris-carneiro** [comment]
Working on AnnotationOnSameLineCheck, AllTestsTest.java fails, because at some point walk()
```java
walk(Paths.get("src/test/java"), filePath -> {
    verifyHasProductionFile(allTests, filePath.toFile());
});
```
is given a .DS_Store as filePath input. 

Since, it is OS specific and I guess it would never happen on CI, though I'm questioning whether an improvement should be considered. 
For now I added a check here as a workaround: 
```java 
private static void verifyHasProductionFile(Map<String, List<String>> allTests, File file) {
    if (file.isFile() && !file.isHidden()) {
        ...
    }
}
```
Similar problem encountered and notified in this [comment](https://github.com/checkstyle/checkstyle/issues/7643#issuecomment-604791314)
What do you guys think ?

**Comment by Mostafahassen1** [comment]
Is this issue open?

**Comment by Zopsss** [comment]
@romani it'd be helpful to have a list of checks which are covered and which aren't, it'll help to track the issue and help contributors to select the next check.

**Comment by romani** [comment]
I put link to latest suppression list that is embedded in source code, there are still bunch of Checks Inputs for update.

Each PR removes one line from suppression.

**Comment by Mostafahassen1** [comment]
I have a quick question regarding the Checkstyle violation comments. If a line exceeds 100 characters, would it be acceptable to use `// violation below` instead of `// violation` in order to stay within the character limit?

**Comment by romani** [comment]
```
public sealed interface InputSealedShouldHavePermitsListTopLevelSealedInterface { 
// violation above 'Sealed classes or interfaces should explicitly declare'

```

**Comment by Mostafahassen1** [comment]
I couldn’t find the answer to my question in the previous PR. What should I do if the maximum line length in the file is 85, but the violation message says it’s 97? I tried splitting it into two lines, but that caused an error."

**violation message**
```java
// violation 'Sealed classes or interfaces should explicitly declare permitted subclasses'
```


**code**

```java
//non-compiled with javac: Compilable with Java17
package com.puppycrawl.tools.checkstyle.checks.design.sealedshouldhavepermitslist;

// xdoc section -- start
public class Example1 {

  // imagine hundreds of lines of code...

  sealed class A {}
  // violation above 'Sealed classes or interfaces should explicitly declare permitted  subclasses'
  final class B extends A {}
  final class C extends A {}
  final class D { } // this can extend A, so as any other class in the same file
}

```

### these solution couldn’t work 
```java
 sealed class A {} // violation 'Sealed classes or interfaces
  // should explicitly declare permitted subclasses'
  final class B extends A {}
  final class C extends A {}
  final class D { } // this can extend A, so as any other class in the same file
}
```
or that 

```java
  // imagine hundreds of lines of code...

  sealed class A {} 
  /* violation above 'Sealed classes or interfaces
   should explicitly declare permitted subclasses'*/
  final class B extends A {}
  final class C extends A {}
  final class D { } // this can extend A, so as any other class in the same file
}


```

**Comment by romani** [comment]
` // violation above 'should explicitly declare permitted subclasses'`

**Comment by Angkitadutta** [comment]
Is this issue open now?

**Comment by Basmalamoustafa** [comment]
Hello @romani, I would love to work on this issue.
Could you please assign it to me? 
Thank you!

**Comment by romani** [comment]
No assignments, just choose Check you like and do comment "I am in xxxxxCheck" and start working on it. Thanks a lot.

**Comment by Angkitadutta** [comment]
I am on it

**Comment by mernaahmed2020** [comment]
"I am in the FileLength Check. I have reviewed the code and ensured that a violation message is added for the case when the file length exceeds the specified limit. I updated the log method to use the appropriate violation message and verified that it now provides clear feedback on the file length violation."

**Comment by Jaykishor-Sharma** [comment]
is this open issue?

**Comment by Volcandrabuzz** [comment]
is the issue still open?

**Comment by romani** [comment]
Yes. 
There more than hundred items to fix 
https://github.com/checkstyle/checkstyle/blob/24eedce6a64145442db5b478c7a39ce0763dad8c/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java#L206

**Comment by budi641** [comment]
Hi @romani is this issue still available?

**Comment by romani** [comment]
Yes, more than hundred of updates are still waiting to be done

**Comment by budi641** [comment]
> Yes, more than hundred of updates are still waiting to be done

great ,ill start working on it.

**Comment by ar-rana** [comment]
> Yes, more than hundred of updates are still waiting to be done

great, I am on the UnnecessarySemicolonAfterOuterTypeDeclaration check.

**Comment by gupta-devang** [comment]
Hi @romani ,
I am picking up ```com.puppycrawl.tools.checkstyle.checks.imports.AvoidStaticImportCheck```

Before I start working on it. What should be the violation message for it?

**Comment by romani** [comment]
https://checkstyle.org/checks/imports/avoidstaticimport.html#Violation_Messages

**Comment by Harsh-Agrawal96** [comment]
I am picking up "coding.HiddenFieldCheck" and start working on it

**Comment by romani** [comment]
Please look around a code base there is `// 2 violations above:` syntax

**Comment by Harsh-Agrawal96** [comment]
ok I find the syntax, I used
// 2 violations above: 
//    ''stringValue' hides a field'
//    ''intValue' hides a field'

but it gives the error 
![Screenshot 2025-01-15 234900](https://github.com/user-attachments/assets/f050ce87-4f38-473d-96b8-a9249bede4d2)

and this same syntax working well with other file which has 2 violations, then what should I do

**Comment by romani** [comment]
watch videos especially  last at [Starting_Development](https://checkstyle.org/beginning_development.html#Starting_Development).

**Comment by smita1078** [comment]
Hi @romani ,
I picked up  "coding.Illegalthrows" , have raised the PR

**Comment by smita1078** [comment]
Hi @romani ,
PR open for "coding.InnerAssignmentsCheck"
Could you please approve..

**Comment by k99sharma** [comment]
Hey, it seems there are still many suppressed checks present in the list. I would like to work on this issue if no one else is working on it.

**Comment by romani** [comment]
there are a lot to update, please send a PR.

**Comment by k99sharma** [comment]
I have already raised 2 PRs, should I fix them all at once and send a single PR or do you want me to raise seperate PRs for each check. Seperate PRs will be easier to review.

**Comment by YuktiNandwana** [comment]
@romani 
is this issue open?

**Comment by ivamly** [comment]
I am on MagicNumberCheck

**Comment by vivek-0509** [comment]
I am on LineLengthCheck

**Comment by vivek-0509** [comment]
I am on NoWhitespaceAfterCheck

**Comment by vivek-0509** [comment]
i am on AvoidEscapedUnicodeCharactersCheckTest

**Comment by vivek-0509** [comment]
I am on ParenPadCheck

**Comment by aclfe** [comment]
**I'm am on IllegalTypeCheck.**

**Comment by romani** [comment]
No
Choose module from https://github.com/checkstyle/checkstyle/blob/1a80f399b16cc6c36697bf2083b2ac70f7ec23d0/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java#L238
Remove it from this collection, run tests, you will get bunch violation in InputXxxxx files, you need to put violation message to such files.

See example of update https://github.com/checkstyle/checkstyle/pull/18160/files

**Comment by aclfe** [comment]
Thanks for double-checking and catching that. I realize I phrased my comment poorly. To clarify, what I meant was deleting the respective module suppressed check from configparser and fixing it's respective violation messages

**Comment by aclfe** [comment]
After much confusion, I manually opened IllegalTypeCheck's tests and found that they've already been done. 
Looking at the changes made last month by someone else: https://github.com/checkstyle/checkstyle/commit/90158f09926eb1f7d89edd55720747e68112f603 (not in this specific PR, but opening an ExampleX.java file we can see the violation messages updated)

Thus, I'll make a simple PR to just remove the suppression line as the changes have already been done

**Comment by aclfe** [comment]
~~**I am on IllegalTokenTextCheck**~~

Free for grabs

**Comment by Youssefalaa7** [comment]
I am on NestedForDepthCheck

**Comment by 0xChaitanya** [comment]
I am on ExplicitInitializationCheck

**Comment by 0xChaitanya** [comment]
is it necessary to have consistency across violation messages for a single check?

```
public class InputExplicitInitialization {
    private int x = 0; // violation 'x' explicitly initialized to '0' (default value)'
    private Object bar = /* comment test */null; // violation 'bar' explicitly initialized to 'null' (default value)'

```

```
public class InputExplicitInitialization {
    private int x = 0; // violation 'x' explicitly initialized to '0' (default value)'
    private Object bar = /* comment test */null; // violation 'bar' explicitly initialized to 'null''

```

I asked this because some lines exceed 100 symbols limit, so I cannot enter more information about the violation in a single line. Because more information makes violation message even clearer to understand.
```
[ERROR] [checkstyle] [ERROR] D:\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\checks\coding\explicitinitialization\InputExplicitInitialization.java:96: Line should not be longer than 100 symbols [lineLength]
```

**Comment by aclfe** [comment]
@0xChaitanya 

> is it necessary to have consistency across violation messages for a single check?

Keep it consistent. To solve your issue, try wrapping the code

**Comment by romani** [comment]
If there is some edge cases, we can make them not consistent, but try to keep consistent.

**Comment by 0xChaitanya** [comment]
@romani @aclfe thanks

**Comment by 0xChaitanya** [comment]
@aclfe when trying to wrap the code I am getting the following error, please suggest some solution
```
public class InputExplicitInitialization {
    private int x = 0; /* violation 'Variable 'x' explicitly initialized to 0 
    (default value for its type)'*/
    private
    Object bar = /* comment test */null; /* violation 'Variable 'bar' explicitly initialized to 'null'
    (default value for its type)*/

... lots of lines of code
```

```
[ERROR] com.puppycrawl.tools.checkstyle.checks.coding.ExplicitInitializationCheckTest.testDefault -- Time elapsed: 0.055 s <<< FAILURE!
Violation lines for D:\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\checks\coding\explicitinitialization\InputExplicitInitialization.java differ.
unexpected (2): 11, 14
---
expected      : [19, 20, 24, 24, 25, 28, 29, 32, 33, 49, 50, 57, 58, 59, 62, 63, 64, 99, 100, 101]
but was       : [11, 14, 19, 20, 24, 24, 25, 28, 29, 32, 33, 49, 50, 57, 58, 59, 62, 63, 64, 99, 100, 101]
        at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.verifyViolations(AbstractModuleTestSupport.java:735)
        at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.verifyWithInlineConfigParser(AbstractModuleTestSupport.java:275)
        at com.puppycrawl.tools.checkstyle.checks.coding.ExplicitInitializationCheckTest.testDefault(ExplicitInitializationCheckTest.java:62)
```

I tried to change the message but the following error occurs:
```
[ERROR] Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.061 s <<< FAILURE! -- in com.puppycrawl.tools.checkstyle.checks.coding.ExplicitInitializationCheckTest
[ERROR] com.puppycrawl.tools.checkstyle.checks.coding.ExplicitInitializationCheckTest.testDefault -- Time elapsed: 0.046 s <<< FAILURE!
Actual and expected violations differ.
expected to match: 11:(?:\d+:)?\s.*'x' explicitly initialized to 0 \(default value for its type\).*
but was          : 11:17: Variable 'x' explicitly initialized to '0' (default value for its type).
        at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.verifyViolations(AbstractModuleTestSupport.java:739)
        at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.verifyWithInlineConfigParser(AbstractModuleTestSupport.java:275)
        at com.puppycrawl.tools.checkstyle.checks.coding.ExplicitInitializationCheckTest.testDefault(ExplicitInitializationCheckTest.java:62)

```

**Comment by aclfe** [comment]
Looking at the error, I see there's a violation message mismatch. 

The "was" has single quotes around 0 and other places. You might need to use the escape character to target those single quotes around different words and characters 

Futhermore, the 

```
expected      : [19, 20, 24, 24, 25, 28, 29, 32, 33, 49, 50, 57, 58, 59, 62, 63, 64, 99, 100, 101]
but was       : [11, 14, 19, 20, 24, 24, 25, 28, 29, 32, 33, 49, 50, 57, 58, 59, 62, 63, 64, 99, 100, 101]
```
shows that errors are being expecting violation strings on the different lines. 

See attached PRs for a reference:
#16116 
#18160

**Comment by 0xChaitanya** [comment]
@aclfe 
I referenced the issue tagged by you, but I am still stuck as the violation message line exceeds 100 characters. Am I allowed to change the messages.properties file so that I can shorten the message?

Here is my new way of doing :
```
public class InputExplicitInitialization {
    // violation below 'Variable 'x' explicitly initialized to '0' (default value for its type)'
    private int x = 0;
    // violation below 'Variable 'bar' explicitly initialized to 'null' (default value for its type)'
    private Object bar = /* comment test */null;
```

```
>>> len("    // violation below 'Variable 'bar' explicitly initialized to 'null' (default value for its type)'")
>>> 101
```

**Comment by aclfe** [comment]
you shouldn't need to modify `messages.properties` for this

You're facing error with the variable 'bar' (length 101 as shown), try changing the name and implementing other methods.

**Comment by 0xChaitanya** [comment]
> you shouldn't need to modify `messages.properties` for this
> 
> You're facing error with the variable 'bar' (length 101 as shown), try changing the name and implementing other methods.

So I am allowed to change the name of variable bar to something else?

@romani

**Comment by aclfe** [comment]
You can give it a try and send the PR. See what romani says.
I'm only recommending a possible solution, not a concrete decision

**Comment by romani** [comment]
please send PR , the best way to resolve issue is to see all changes that you have on your local - just make PR.
It is completely ok for PR to be not perfect from first commit. 

you can use partial message in long lines, or use "// violation above" or "// violation below"

**Comment by Adam-Gamil** [comment]
I am on IllegalTokenTextCheck

**Comment by Avanish-Gupta-CSE** [comment]
I'm on it - NestedTryDepthCheck (PR #18779 already submitted)

**Comment by Avanish-Gupta-CSE** [comment]
I'm on it - NestedTryDepthCheck

**Comment by Adam-Gamil** [comment]
I am on MatchXpathCheck

**Comment by MostafaAbdulazziz** [comment]
i am on missingOverrideCheck

**Comment by MostafaAbdulazziz** [comment]
I am on StringLiteralEqualityCheck

**Comment by MostafaAbdulazziz** [comment]
I am on SuperFinalizeCheck

**Comment by lithops-zty** [comment]
I am on NonEmptyAtclauseDescriptionCheck and JavadocMissingWhitespaceAfterAsteriskCheck

**Comment by yn-mi12** [comment]
Hello, I can work on this if the issue is still open.

**Comment by surajgojanur** [comment]
I would like to work on removing ParameterNameCheck from SUPPRESSED_CHECKS as part of this issue.

I will prepare a focused PR for it.

**Comment by developer4949-code** [comment]
Hi @romani, I'm a GSoC 2026 applicant. I'd like to work on adding violation messages for a Check under issue #15456. Which checks still need updating? I'm ready to submit a PR today.

**Comment by surajgojanur** [comment]
I'm on HideUtilityClassConstructorCheck

**Comment by bhumika-chaudhari** [comment]
I am in ExplicitInitializationCheck

**Comment by HamzehAdawi** [comment]
I am on  RegexpMultilineCheck

**Comment by Avanish-Gupta-CSE** [comment]
Working on DesignForExtensionCheck for Issue #15456.

Will add explicit violation messages to improve developer experience. This helps make code violations clearer and more understandable for users.

**Comment by Avanish-Gupta-CSE** [comment]
Switching to InnerTypeLastCheck for Issue #15456.

DesignForExtensionCheck has too many violation instances across multiple files. InnerTypeLastCheck should be more manageable for adding explicit violation messages to improve developer experience.

**Comment by Avanish-Gupta-CSE** [comment]
Working on OneTopLevelClassCheck for Issue #15456.

This check ensures each file has only one top-level class. Will add explicit violation messages to improve developer experience and make the violations clearer.

**Comment by developer4949-code** [comment]
Hi @romani,
I’m working on issue #15456. I’ll take one check, add explicit violation messages to its inline violation comments, and send a focused PR for that check.

**Comment by romani** [comment]
issue still valid, we found defect in validation, fixed it, bunch of problems are leaked, so numerous update still required, all items to update are in suppression list, choose one and help us to fix it.

**Comment by NewRecon** [comment]
Hi! I'm on `RightCurlyCheck`

**Comment by anuruddha-pratap-singh** [comment]
Hello!
I am on it — taking SimplifyBooleanExpressionCheck

**Comment by r-afael** [comment]
I'm working on MethodCountCheck: adding the missing violation messages to its test inputs and removing its SUPPRESSED_CHECKS entry.
